src/Microsoft.Health.Fhir.Api/Features/Routing/SearchPostReroutingMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Health.Fhir.Api.Features.Routing
+{
+    public class SearchPostReroutingMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public SearchPostReroutingMiddleware(RequestDelegate next)
+        {
+            EnsureArg.IsNotNull(next);
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var request = context.Request;
+
+            if (request != null
+                && request.Method == "POST"
+                && request.Path.Value.EndsWith(KnownRoutes.Search, System.StringComparison.OrdinalIgnoreCase))
+            {
+                if (request.ContentType is null || request.ContentType == "application/x-www-form-urlencoded")
+                {
+                    if (request.HasFormContentType)
+                    {
+                        var dic = request.Query.ToDictionary(k => k.Key, v => v.Value);
+                        foreach (var elem in request.Form)
+                        {
+                            dic.Add(elem.Key, elem.Value);
+                        }
+
+                        request.Query = new QueryCollection(dic);
+                    }
+
+                    request.Path = request.Path.Value.Substring(0, request.Path.Value.Length - KnownRoutes.Search.Length);
+                    request.Method = "GET";
+                }
+                else
+                {
+                    context.Response.Clear();
+                    context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    await context.Response.WriteAsync(Resources.ContentTypeFormUrlEncodedExpected);
```

Comment:
I think we should have an operation outcome wrapping the error message like we do elsewhere. Example: Post to /Patient with a Content-Type of application/text

Reply:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/index/_static/middleware-pipeline.svg?view=aspnetcore-3.1
![image](https://user-images.githubusercontent.com/1523833/98611284-81a63100-22a6-11eb-90ed-ce24b8296002.png)

I'm using custom middleware which is happening before request hitting endpoint.
So far most of exception conversion happens inside filter on top of controller/endpoint.

Right now main reason why I don't return OperationOutcome is because it's not accessible for me here.
I include this middleware to non-spec specific project, and operation outcome is part of standard.

Considering it's a bit hard for me to figure out how to make it work, I would argue what we don't have operation outcome result for every error, there is still plenty of 404 or 401 which are not wrapped I believe.